.github/workflows/main.yml

```diff
@@ -105,3 +105,32 @@ jobs:
         git commit -m "[CI] Update staging for ${DVER} on ${GH_BRANCH}" || true
         
         git push origin main || true
+
+  # docfx metadata generation can often take several minutes, so we run it in parallel to the main build
+  generate-docfx-metadata:
+    name: Generate docfx metadata
+    runs-on: windows-2022
```

Comment:
Do we need to run this task on a windows runner (other than the fact that we already use it)? They're notably slower IME than their Linux alternatives.